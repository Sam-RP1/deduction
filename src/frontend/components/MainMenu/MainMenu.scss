@import '../../styles/partials/color-themes';
@import '../../styles/mixins/transition';

%flex-center-column {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.main-menu {
  @extend %flex-center-column;
  height: calc(100vh - 32px - 80px);;
  width: 100%;

  &__title {
    margin: 0;
    font-size: 5em;
    font-weight: 400;
    @include themed() {
      color: t($text);
    }
    @include transition(color 0.4s ease);
  }

  &__nav-btns-container {
    @extend %flex-center-column;
    margin-top: 30px;

    div {
      &:nth-of-type(n) {
        margin-top: 15px;
      }
      &:nth-of-type(1) {
        margin-top: 0;
      }
    }
  }
}
