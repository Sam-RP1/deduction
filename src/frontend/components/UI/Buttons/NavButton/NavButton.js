import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import '../Button.scss';

const NavButton = (props) => {
    return (
        <NavLink
            to={props.route}
            exact={true}
            onClick={() => {
                props.function();
            }}
        >
            <button className={'btn ' + props.opClasses}>{props.title}</button>
        </NavLink>
    );
};

NavButton.propTypes = {
    title: PropTypes.string.isRequired,
    route: PropTypes.string.isRequired,
    function: PropTypes.func.isRequired,
    opClasses: PropTypes.string,
};

NavButton.defaultProps = {
    title: 'UNDEFINED',
    route: '/undefined',
    function: () => console.log('[BUTTON] undefined button function'),
    opClasses: '',
};

export default NavButton;
