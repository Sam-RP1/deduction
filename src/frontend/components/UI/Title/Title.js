import React from 'react';
import PropTypes from 'prop-types';

import './Title.scss';

const Title = (props) => {
    return <h1 className='title'>{props.title}</h1>;
};

Title.propTypes = {
    title: PropTypes.string,
};

Title.defaultProps = {
    title: 'Deduction',
};

export default Title;
