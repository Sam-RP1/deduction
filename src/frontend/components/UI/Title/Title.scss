@import '../../../styles/partials/color-themes';
@import '../../../styles/mixins/transition';

.title {
  height: 90px;
  margin: 0;
  font-size: 5em;
  font-weight: 400;
  @include themed() { color: t($text); }
  @include transition-theme(color 0.4s ease);
  @media screen and (max-width: 350px) { font-size: 4.5em; }
}
