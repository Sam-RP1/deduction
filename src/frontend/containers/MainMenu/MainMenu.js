import React from 'react';

import PresMainMenu from '../../components/Menus/MainMenu/MainMenu';

const MainMenu = () => {
    const menuButtons = [
        { title: 'Create Game', route: '/creategame', function: () => console.log('[MAIN MENU BTN] create game') },
        { title: 'Join Game', route: '/joingame', function: () => console.log('[MAIN MENU BTN] join game') },
        { title: 'Rules', route: '/rules', function: () => console.log('[MAIN MENU BTN] rules') },
    ];

    return <PresMainMenu buttons={menuButtons} />;
};

export default MainMenu;
