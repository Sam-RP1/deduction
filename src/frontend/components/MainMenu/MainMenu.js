import React, { useEffect } from 'react';
import Typed from 'typed.js';

import NavButton from '../UI/Buttons/NavButton/NavButton';

import '../../styles/root.scss';
import './MainMenu.scss';

const MainMenu = () => {
  const menuButtons = [
    { title: "Start Game", route: "/startgame", function: () => console.log("TEMP BTN FUNC") },
    { title: "Join Game", route: "/joingame", function: () => console.log("TEMP BTN FUNC") },
    { title: "Rules", route: "/rules", function: () => console.log("TEMP BTN FUNC") }
  ]

  useEffect(() => {
    const typedOptions = {
      strings: ["Deduction"],
      startDelay: 300,
      typeSpeed: 90,
      showCursor: false
    };

    const elem = document.getElementsByClassName("main-menu__title")[0];

    const typed = new Typed(elem, typedOptions); // eslint-disable-line
  }, [])

  return (
    <section className="main-menu">
    <h1 className="main-menu__title"></h1>
    <div className="main-menu__nav-btns-container">
    {menuButtons.map((item) => {
      return (
        <NavButton key={item.route} title={item.title} route={item.route} function={item.function} />
      )
    })}
    </div>
    </section>
  )
}

export default MainMenu;
