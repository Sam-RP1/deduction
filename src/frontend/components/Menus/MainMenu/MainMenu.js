import React from 'react';
import PropTypes from 'prop-types';

// Imports
import Title from '../../UI/Title/Title';
import NavButton from '../../UI/Buttons/NavButton/NavButton';

// Styles
import './MainMenu.scss';

// Presentational Component
const MainMenu = (props) => {
    return (
        <section className='main-menu'>
            <Title title='Deduction' />
            <div className='main-menu__content'>
                {props.buttons.map((btn, i) => {
                    return <NavButton key={i} title={btn.title} route={btn.route} function={btn.function} />;
                })}
            </div>
        </section>
    );
};

MainMenu.propTypes = {
    buttons: PropTypes.array.isRequired,
};

MainMenu.defaultProps = {
    buttons: [{ title: 'Error', route: '/', function: () => console.log('[MAIN MENU BTN] error') }],
};

export default MainMenu;
