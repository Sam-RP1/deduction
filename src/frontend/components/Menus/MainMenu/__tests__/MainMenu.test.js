import React from 'react';

// Enzyme Imports
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

// Component Imports
import MainMenu from '../MainMenu';
import Title from '../../../UI/Title/Title';
import NavButton from '../../../UI/Buttons/NavButton/NavButton';

// Enzyme configuration
configure({ adapter: new Adapter() });

// Test cases
describe('<MainMenu /> | Component', () => {
    const dummyData = [
        { title: 'Create Game', route: '/creategame', function: () => console.log('[MAIN MENU BTN] create game') },
        { title: 'Join Game', route: '/joingame', function: () => console.log('[MAIN MENU BTN] join game') },
        { title: 'Rules', route: '/rules', function: () => console.log('[MAIN MENU BTN] rules') },
    ];

    it('should render <Title />', () => {
        const wrapper = shallow(<MainMenu />);
        expect(wrapper.find(<Title />));
    });

    it('should render one <NavButton />', () => {
        const wrapper = shallow(<MainMenu buttons={[dummyData[0]]} />);
        expect(wrapper.find(<NavButton />));
    });

    it('should render three <NavButton />', () => {
        const wrapper = shallow(<MainMenu buttons={dummyData} />);
        expect(wrapper.find('.main-menu__content').children().length).toEqual(3);
    });

    it('should render <Title /> and three <NavButton />', () => {
        const wrapper = shallow(<MainMenu buttons={dummyData} />);
        expect(wrapper.find(<Title />));
        expect(wrapper.find('.main-menu__content').children().length).toEqual(3);
    });

    it('should render correctly', () => {
        const wrapper = shallow(<MainMenu buttons={dummyData} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
