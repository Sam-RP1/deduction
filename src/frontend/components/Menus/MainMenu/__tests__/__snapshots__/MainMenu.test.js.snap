// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> | Component should render correctly 1`] = `
<section
  className="main-menu"
>
  <Title
    title="Deduction"
  />
  <div
    className="main-menu__content"
  >
    <NavButton
      function={[Function]}
      key="0"
      opClasses=""
      route="/creategame"
      title="Create Game"
    />
    <NavButton
      function={[Function]}
      key="1"
      opClasses=""
      route="/joingame"
      title="Join Game"
    />
    <NavButton
      function={[Function]}
      key="2"
      opClasses=""
      route="/rules"
      title="Rules"
    />
  </div>
</section>
`;
